core/src/main/scala/dagr/core/tasksystem/Task.scala

```diff
@@ -23,20 +23,167 @@
  */
 package dagr.core.tasksystem
 
-import com.fulcrumgenomics.commons.CommonsDef.unreachable
+import java.time.{Duration, Instant}
+
+import com.fulcrumgenomics.commons.CommonsDef.{FilePath, unreachable}
+import com.fulcrumgenomics.commons.io.PathUtil
+import com.fulcrumgenomics.commons.util.TimeUtil.formatElapsedTime
+import dagr.api.models.{ResourceSet, TaskStatus, TimePoint}
+import dagr.core.DagrDef.TaskId
+import dagr.core.exec.Executor
 import dagr.core.execsystem.TaskExecutionInfo
+import dagr.core.tasksystem.Task.TaskInfo
 
 import scala.collection.mutable
 import scala.collection.mutable.ListBuffer
+import scala.reflect.ClassTag
 import scala.util.control.Breaks._
 
 /** Utility methods to aid in working with a task. */
 object Task {
 
+  trait TaskInfoLike extends dagr.api.models.TaskInfoLike with Ordered[TaskInfoLike] {
```

Comment:
The base trait for `TaskInfo`s for `dagr.core`. Added to make it ordered, add in both a `scriptPath` and `logPath` method, and finally to store a reference to the actual task (good luck getting the latter to compile in scala-js).